caikit/runtime/client/remote_model_finder.py

```diff
@@ -0,0 +1,295 @@
+            )
+
+            # Construct secure channel
+            channel_func = grpc.secure_channel
+            channel_args = [target, grpc_credentials, options]
+        else:
+            channel_func = grpc.insecure_channel
+            channel_args = [target, options]
+
+        # Construct Info RPC and submit request
+        log.debug2(
+            "Constructing grpc finder channel with %s and %s",
+            channel_func,
+            str(channel_args),
+        )
+        with channel_func(*channel_args) as channel:
+            info_service_rpc = channel.unary_unary(
+                get_grpc_route_name(ServiceType.INFO, "GetModelsInfo"),
+                request_serializer=ModelInfoRequest.get_proto_class().SerializeToString,
+                response_deserializer=ModelInfoResponse.get_proto_class().FromString,
+            )
+            try:
+                model_info_proto = info_service_rpc(
+                    ModelInfoRequest().to_proto(), timeout=self._connection.timeout
+                )
+            except grpc.RpcError as exc:
+                log.warning(
+                    "Unable to discover modules from remote: %s. Error: %s",
+                    self._connection.hostname,
+                    str(exc),
+                )
+                return {}
+
+        model_info_response = ModelInfoResponse.from_proto(model_info_proto)
+
+        # Parse response into dictionary of name->id
+        supported_modules = {}
+        for model_info in model_info_response.models:
+            model_name = model_info.name
+            module_id = model_info.module_id
+
+            log.debug(
+                "Discovered model %s with module_id %s from remote runtime",
+                model_name,
+                module_id,
+            )
+            supported_modules[model_name] = module_id
+
+        return supported_modules
+
+    def _discover_http_models(self) -> Dict[str, str]:
+        """Helper function to get all the supported models and modules
+        from a remote HTTP runtime
+
+        Returns:
+            supported_models:Dict[str, str]
+                Mapping of remote model names to module_ids
+        """
+
+        # Configure HTTP Client object
+        target = (
+            f"{self._connection.hostname}:{self._connection.port}{MODELS_INFO_ENDPOINT}"
+        )
+        request_kwargs = {}
+        if self._tls.enabled:
+            target = f"https://{target}"
+
+            # Configure the TLS CA settings
+            if self._tls.insecure_verify:
+                request_kwargs["verify"] = False
+            else:
+                if self._tls.ca_file:
+                    request_kwargs["verify"] = self._tls.ca_file
+                else:
+                    request_kwargs["verify"] = True
+
+            if self._tls.cert_file and self._tls.key_file:
+                request_kwargs["cert"] = (self._tls.cert_file, self._tls.key_file)
```

Comment:
Ah, barf. It took some spelunking, but I can confirm that `requests` [does require files](https://github.com/psf/requests/blob/main/src/requests/adapters.py#L276).

Reply:
Given this, I wonder if it's worth exploring using a `requests.Session` to avoid reading these files from disk on every request?

Reply:
Good call! I updated to use `requests.Session`